win32/build/confutils.js

```diff
@@ -2676,27 +2676,101 @@ function toolset_setup_common_libs()
 }
 
+function vs_warnings() {
+    var warnings = {};
+    warnings[4100] = "disable";	// Unreferenced formal parameter
+    warnings[4127] = "disable";	// Conditional expression is constant (usually a macro expansion, not an error)
+
+    warnings[4668] = "disable";	// Using #if UNDEFINED_MACRO instead of #ifdef UNDEFINED_MACRO
+    warnings[4710] = "disable";	// The compiler decided not to inline a function marked as INLINE
+    warnings[4711] = "disable";	// The compiler decided to inline a function
+
+    warnings[4820] = "disable";	// Message about padding bytes added to a structurenote
+
+    // By default, output all warnings but not as errors.
+    ADD_FLAG("CFLAGS", "/Wall /WX-");
+
+    // turn the warning array into compiler switches
+    for (var k in warnings) {
+        switch (warnings[k]) {
+            case "disable":
+                ADD_FLAG("CFLAGS", "/wd" + k);
+                break;
+            case "once":
+                ADD_FLAG("CFLAGS", "/wo" + k);
+                break;
+            case "error":
+                ADD_FLAG("CFLAGS", "/we" + k);
+                break;
+            default:
+                ERROR("Unknown warning type '" + warnings[k] + "' for warning " + k);
+                break;
+        }
+    }
+
+    // also perform static analysis of the code and output it to the build directory
+    ADD_FLAG("CFLAGS", "/analyze:log \"$(BUILD_DIR)\\codeanalysis.txt\" /analyze:quiet ");
+}
+
 function toolset_setup_build_mode()
 {
-	if (PHP_DEBUG == "yes") {
-		ADD_FLAG("CFLAGS", "/LDd /MDd /W3 /Gm /Od /D _DEBUG /D ZEND_DEBUG=1 " +
-			(X64?"/Zi":"/ZI"));
-		ADD_FLAG("LDFLAGS", "/debug");
-		// Avoid problems when linking to release libraries that use the release
-		// version of the libc
-		ADD_FLAG("PHP_LDFLAGS", "/nodefaultlib:msvcrt");
-	} else {
-		// Generate external debug files when --enable-debug-pack is specified
-		if (PHP_DEBUG_PACK == "yes") {
-			ADD_FLAG("CFLAGS", "/Zi");
-			ADD_FLAG("LDFLAGS", "/incremental:no /debug /opt:ref,icf");
-		}
-		// Equivalent to Release_TSInline build -> best optimization
-		ADD_FLAG("CFLAGS", "/LD /MD /W3 /Ox /D NDebug /D NDEBUG /D ZEND_WIN32_FORCE_INLINE /GF /D ZEND_DEBUG=0");
-
-		// if you have VS.Net /GS hardens the binary against buffer overruns
-		// ADD_FLAG("CFLAGS", "/GS");
-	}
+    if (PHP_DEBUG == "yes") {
+        ADD_FLAG("CFLAGS", "/LDd /MDd /Gm /Od /D _DEBUG /D ZEND_DEBUG=1");
+        // Avoid problems when linking to release libraries that use the release
+        // version of the libc
+        ADD_FLAG("PHP_LDFLAGS", "/nodefaultlib:msvcrt");
+
+        // Poison uninitialized locals, detect local variable stack buffer under/overruns and detect calling-convention errors
+        ADD_FLAG("CFLAGS", "/RTCs");
+
+        // Detect integer truncation errors
+        ADD_FLAG("CFLAGS", "/RTCc");
+
+        // Abort on uninitialized local use
+        ADD_FLAG("CFLAGS", "/RTCu");
+

```

Comment:
About? The warnings things?
On Jul 16, 2015 9:00 PM, "Nikita Popov" <notifications@github.com> wrote:

> In win32/build/confutils.js
> <https://github.com/php/php-src/pull/1418#discussion_r34789510>:
>
> > -	}
> > +    if (PHP_DEBUG == "yes") {
> > +        ADD_FLAG("CFLAGS", "/LDd /MDd /Gm /Od /D _DEBUG /D ZEND_DEBUG=1");
> > +        // Avoid problems when linking to release libraries that use the release
> > +        // version of the libc
> > +        ADD_FLAG("PHP_LDFLAGS", "/nodefaultlib:msvcrt");
> > +
> > +        // Poison uninitialized locals, detect local variable stack buffer under/overruns and detect calling-convention errors
> > +        ADD_FLAG("CFLAGS", "/RTCs");
> > +
> > +        // Detect integer truncation errors
> > +        ADD_FLAG("CFLAGS", "/RTCc");
> > +
> > +        // Abort on uninitialized local use
> > +        ADD_FLAG("CFLAGS", "/RTCu");
> > +
>
> @pierrejoye <https://github.com/pierrejoye> Do you have a link handy to
> that internals discussion?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1418/files#r34789510>.
>